mteb/abstasks/languages.py

```diff
@@ -14,3 +17,54 @@
 
 with path_to_lang_scripts.open("r") as f:
     ISO_TO_SCRIPT = json.load(f)
+
+
+@dataclass
+class LangScriptFilter:
+    language_scripts: set[str]
+    scripts: set[str]
+    languages: set[str]
+
+    @classmethod
+    def from_languages_and_scripts(
+        cls, languages: list[str] | None, scripts: list[str] | None
+    ) -> LangScriptFilter:
+        lang_script_codes = set()
+        # normalize to 3 letter language codes
+        normalized_langs = set()
+        filter_lang = languages is not None
+
+        if filter_lang:
+            for lang in languages:
+                lang_script = lang.split("-")
+
+                is_lang_script_code = len(lang_script) == 2
+                if is_lang_script_code:
```

Comment:
It is a way to mark the intent of `len(lang_script) == 2` which might otherwise be unclear. In that way, it acts more like a comment (you might be interested in this [video argument](https://www.youtube.com/watch?v=Bf7vDBBOBUA&t=168s))

Reply:
I have to agree with @isaac-chung here. Especially with the `filter_lang = languages is not None` line. I had to go back and check what that variable contained and it was a bit hard for me to follow the flow of the code. Naming things in a way that everyone understands is way harder than letting people read the behaviour directly and maybe including a comment where the intent isn't clear.

Reply:
fixed